### PR TITLE
Add fpm package manifest

### LINKED ISSUE #17: functional-fortran as an fpm package
Hello Milan, a lot of thanks for all your contributions. I think it must be very useful to have the functional-fortran as an fpm package.

### PR DESCRIPTION
**Pull request by awvwgk**
- support usage of functional-fortran with fpm
- does not support testing due to shortcomings of bootstrap fpm

Fixes #17

### CODE REVIEW

**Comment by milancurcic** [comment]
Thanks a lot @awvwgk. Did you check that `fpm build` works?

**Comment by awvwgk** [comment]
Of course, checked with bootstrap fpm and GCC 10.2.

**Comment by milancurcic** [comment]
Great, thank you. 

One more thing: I think the package name should be just "functional". Rationale: The repo name "functional-fortran", like with "datetime-fortran" was a branding decision. It allowed it to stand out, and, putting words "functional" and "Fortran" together helped it get attention. However, in the context of fpm and Fortran packages, the word "fortran" is redundant. 

I did the same with [datetime-fortran](https://github.com/wavebitscientific/datetime-fortran/blob/master/fpm.toml), which as an fpm package is just `datetime`. 

What do you think?

File: fpm.toml [hunk 1]
```diff
@@ -1,6 +1,9 @@
-name = "functional-fortran"
+name = "functional"
 version = "0.5.0"
 license = "BSD-3-Clause"
+author = "Milan Curcic"
+maintainer = "mcurcic@wavebitscientific.com"
+copyright = "Copyright (c) 2016-2020, Milan Curcic"
```
Comment (milancurcic) [inline fpm.toml]:
```suggestion
copyright = "Copyright (c) 2016-2020, functional-fortran contributors"
```

Unless you don't accept the copyright being the only other contributor :)

Comment (awvwgk) [reply fpm.toml]: Not really a fan of the author/copyright/maintainer entries in the package manifest, see https://github.com/fortran-lang/fpm-registry/issues/24. Makes it even harder to keep all the meta data in the repo consistent.

In this case the fpm.toml would mismatch the header in the BSD-3-Clause license:
https://github.com/wavebitscientific/functional-fortran/blob/f9baf451a3256c84165c2b9a3fdffd335ba1124e/LICENSE#L1

Comment (milancurcic) [reply fpm.toml]: Yes, the LICENSE file should as well be updated in this PR.

I tend to agree, let's discuss it there. Here, feel free to remove or keep any of these optional fields. I don't think they're essential, nor do they bother me.

**Comment by milancurcic** [comment]
Thank you!

**Comment by certik** [comment]
Yes, the name should be just `datetime` and `functional`.

**Pull request merged**